supplier/tests/test_supplier_page.py

```diff
@@ -39,3 +93,77 @@ def test_supplier_check_false(non_supplier_mock):
 def test_supplier_check_true(saved_supplier0):
     assert saved_supplier0 in Supplier.objects.all()
     assert get_supplier(saved_supplier0) == saved_supplier0
+
+
+@pytest.mark.django_db()
+def test_get_product_return_supplier_product(request_searched_product_mock,
+                                             saved_supplier0,
+                                             saved_supplier_product0,
+                                             saved_supplier_product1,
+                                             searched_product_string):
+    assert saved_supplier_product0.user_name == saved_supplier0
+    assert saved_supplier_product1.user_name == saved_supplier0
+    request_searched_product_mock.GET[searched_product_string] = saved_supplier_product0.qr_code.product_name
+    result = get_product_by_name_and_supplier(request_searched_product_mock, saved_supplier0)
+    assert result.first() == saved_supplier_product0
+
+
+@pytest.mark.django_db()
+def test_get_product_sup_dont_have_product(request_searched_product_mock,
+                                           saved_supplier0,
+                                           saved_product0,
+                                           searched_product_string):
+    assert saved_product0 in Product.objects.all()
```

Comment:
Why are we using here `saved_product0` instead of `saved_supplier_product0`? 
A product isn't connected directly to a supplier. That's why we have the `supplier_product` model.